pymc3/glm/glm.py

```diff
@@ -1,23 +1,41 @@
 import numpy as np
 from ..distributions import Normal
-from ..tuning.starting import find_MAP
 from ..model import modelcontext
 import patsy
-import theano
 import pandas as pd
-from collections import defaultdict
-from pandas.tools.plotting import scatter_matrix
+import theano
+from collections import defaultdict, namedtuple
 
 from . import families
 
 __all__ = ['glm', 'linear_component', 'plot_posterior_predictive']
 
 
-def linear_component(formula, data, priors=None,
-                     intercept_prior=None,
-                     regressor_prior=None,
-                     init_vals=None, family=None,
-                     model=None):
+def _xy_to_data_and_formula(X, y):
+    if not isinstance(y, pd.Series):
+        y = pd.Series(y, name='y')
+    else:
+        if not y.name:
+            y.name = 'y'
+    if not isinstance(X, (pd.DataFrame, pd.Series)):
+        if len(X.shape) > 1:
+            cols = ['x%d' % i for i in range(X.shape[1])]
+        else:
+            cols = ['x']
+        X = pd.DataFrame(X, columns=cols)
+    elif isinstance(X, pd.Series):
+        if not X.name:
+            X.name = 'x'
+    # else -> pd.DataFrame -> ok
+    data = pd.concat([y, X], 1)
+    formula = patsy.ModelDesc(
+        [patsy.Term([patsy.LookupFactor(y.name)])],
+        [patsy.Term([patsy.LookupFactor(p)]) for p in X.columns]
+    )
+    return data, formula
+
+
+class linear_component(namedtuple('Estimate', 'y_est,coeffs')):
```

Comment:
It is a tuple subclass with special constructor. Before you had a function that returned a tuple, now you had a tuple that can be constructed in two ways. Call result is still tuple and thus compatible with old API